### ISSUE TITLE
[REQUEST] Support executing Windows binaries

### ISSUE THREAD

**Issue by phuclv90** [compiler-explorer/compiler-explorer#1879]
CE has wine to run cl.exe, so it should also be possible to run Windows executables after compiling

This will also help running C++/CLI binaries after installing .NET Core or mono. Currently there are very few places that you can try C++/CLI code online

**Comment by apmorton** [comment]
The wine compilers don't even currently support linking (I have never looked into this myself, but I am told there are some non-trivial issues there).

In addition, with our current sandboxing setup all wine compilers run in a single shared sandbox due to complexities of wine.

Enabling user code execution in that environment is a non-starter - there is simply no way we could make it secure enough to prevent malicious abuse.

For this to be a reality we would need to make some pretty dramatic changes in how we sandbox wine processes, and solve some difficult performance problems with cold-booting a wine complex for every user execution.

TL;DR; Never say never, but I don't think its feasible with the current reality of how wine works on CE.

**Comment by mattgodbolt** [comment]
Agreed with wine here. There would need to be a first-class Windows sandbox support from Microsoft for this. (CC @strega-nil in case she has any commentary)

**Comment by phuclv90** [comment]
Does CE have any Windows instances? Windows have sandbox feature from build 1903 which may help

**Comment by strega-nil** [comment]
This is not currently a feature that has had enough ask from our community to allocate the necessary engineer time and political capital to get it working; continue upvoting https://developercommunity.visualstudio.com/idea/663829/allow-code-execution-on-godboltorg-for-msvc-compil.html :)

**Comment by mattgodbolt** [comment]
Thanks @strega-nil :)

**Comment by partouf** [comment]
This was officially put on the Roadmap last November according to https://developercommunity.visualstudio.com/comments/1252353/view.html

**Comment by mattgodbolt** [comment]
Hooray! Will also mention when I next speak with our MS friends.

**Comment by noita-player** [comment]
could you just fix the bug where msvc compilers, under output, have "execute the code" checked. it's not clear from the ux that godbolt can't execute windows bins (of course it's clear if you're an expert and understand how it works, but I'm usually linking godbolt to newbies).

leaving it unchecked and still grayed out will also just result in user confusion... "why is this grayed out?" but, I guess, better than the current false implication

**Comment by partouf** [comment]
> could you just fix the bug where msvc compilers, under output, have "execute the code" checked. it's not clear from the ux that godbolt can't execute windows bins (of course it's clear if you're an expert and understand how it works, but I'm usually linking godbolt to newbies).

I'm confused, is it not grayed out?

**Comment by RubenRBS** [comment]
I've checked the config for the MSVC compilers and they don't report as being able to execute code. Could you give us a link that shows this?

**Comment by timmy-ms** [comment]
I am working on enabling this currently, but we have some outstanding questions about a open source tool we need to include.

**Comment by davispuh** [comment]
This is so confusing, so is this supported or not?

In past it wasn't supported, but then this MS blog says it's supported https://devblogs.microsoft.com/cppblog/execution-and-static-analysis-support-for-msvc-on-compiler-explorer/

But this is what I see

![attels](https://github.com/compiler-explorer/compiler-explorer/assets/651800/ed0d58ed-14b9-4595-87f6-2e6e842591a0)

**Comment by mattgodbolt** [comment]
This is not currently supported for Microsoft compilers. We do not have a license to run the Microsoft compilers and execution, and so we proxy requests to their infrastructure. For various reasons they had to disable execution support on their side. We are working to try and get a license to run the Microsoft compilers ourselves, at which time we could re-enable execution.

I can't speak for Microsoft's confusing messaging here, unfortunately.
